src/main/java/org/example/Main.java

```diff
@@ -1,7 +1,220 @@
+                }
+
+                if (deleted || taskFound) {
+                    break;
+                }
+
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid input. Please enter a valid numerical ID.");
+            }
+        }
+    }
+
+    private static void listTasks() {
+        if (tasks.isEmpty()) {
+            System.out.println("Your list is empty. Add new task with 'add' command");
+        } else {
+            for (Task task : tasks) {
+                System.out.println(task.toString());
+            }
+        }
+    }
+
+    private static void viewTask() {
+        boolean exists = false;
+        System.out.println("Type ID of the task");
+        int input = scanner.nextInt();
+        for (Task task : tasks) {
+            if (task.getId() == input) {
+                System.out.println("Task info:");
+                System.out.println(task);
+                exists = true;
+                break;
+            }
+        }
+
+        if (!exists) {
+            System.out.println("Task with ID '" + input + "' not found. Use list to view all your tasks");
+        }
+
+        scanner.nextLine();
+    }
+
+    private static void saveTasks() {
+        Path path = Paths.get(SAVE_FILE);
+        List<String> lines = tasks.stream()
+                .map(task -> task.getName() + "," + task.getPriority() + "," + task.getDueDate())
+                .collect(Collectors.toList());
+        try {
+            Files.write(path, lines, StandardCharsets.UTF_8);
+            System.out.println("List has been saved to " + SAVE_FILE);
+            scanner.close();
+        } catch (IOException e) {
+            System.out.println("Error has occurred while saving tasks: " + e.getMessage());
+        }
+
+        running = false;
+    }
+
+    private static void loadTasks() {
+        Path path = Paths.get(SAVE_FILE);
+        if (!Files.exists(path)) {
+            System.out.println("No save file found. Starting with an empty list.");
+            System.out.println("If you think this is a mistake, make sure there is a file in the correct directory.");
+            return;
+        }
+
+        try {
+            List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+            for (String line : lines) {
+                String[] parts = line.split(",");
+                if (parts.length == 3) {
+                    String description = parts[0];
+                    Priority priority = Priority.valueOf(parts[1].toUpperCase());
+                    LocalDate dueDate = LocalDate.parse(parts[2], DateTimeFormatter.ISO_LOCAL_DATE);
+                    tasks.add(new Task(description, priority, dueDate));
+                }
+            }
+            System.out.println("Tasks have been loaded.");
+        } catch (IOException e) {
+            System.out.println("Error has occurred while loading tasks: " + e.getMessage());
```

Comment:
IMPROVE: doporucuji podrobnejsi logovani vyjimek i ze stack tracem a info o tom ze bude pouzit prazdy seznam ukolu nebo ukoncit aplikaci s chybovym kodem.

```suggestion
            System.err.println("Error has occurred while loading tasks: " + e.getMessage());
            System.err.println("Starting with an empty list.");
            e.printStackTrace(System.err);
```